pkg/controller/utils/platform.go

```diff
@@ -0,0 +1,43 @@
+package utils
```

Comment:
Util packages have tendency to hide wrong package structuring and I think we should avoid it .

`pkg/controller/platform` seems like a better fit for this package. Then you could have platform.Type func part of the api.

Reply:
yes, that's WIP and an intermediate step to refactor the existing unflexible AWS-only implementation